packages/server/test/e2e/4_downloads_spec.ts

```diff
@@ -1,14 +1,58 @@
-import e2e from '../support/helpers/e2e'
+import path from 'path'
+
+import e2e, { expect } from '../support/helpers/e2e'
 import Fixtures from '../support/helpers/fixtures'
+import fs from '../../lib/util/fs'
+
+const downloadsProject = Fixtures.projectPath('downloads')
 
 describe('e2e downloads', () => {
   e2e.setup()
 
   e2e.it('handles various file downloads', {
-    project: Fixtures.projectPath('downloads'),
-    spec: '*',
+    project: downloadsProject,
+    spec: 'downloads_spec.ts',
     config: {
       video: false,
     },
   })
+
+  it('trashes downloads between runs', async function () {
+    await e2e.exec(this, {
+      project: downloadsProject,
+      spec: 'download_csv_spec.ts',
+    })
+
+    // this run should trash the downloads from the above run
+    await e2e.exec(this, {
+      project: downloadsProject,
+      spec: 'simple_passing_spec.ts',
+    })
+
+    const filePath = path.join(downloadsProject, 'cypress', 'downloads', 'records.csv')
+    const exists = await fs.pathExists(filePath)
+
+    expect(exists, `Expected ${filePath} not to exist, but it does`).to.be.false
+  })
+
+  it('does not trash downloads between runs if trashAssetsBeforeRuns: false', async function () {
+    await e2e.exec(this, {
+      project: downloadsProject,
+      spec: 'download_csv_spec.ts',
+    })
+
+    // this run should _not_ trash the downloads from the above run
+    await e2e.exec(this, {
+      project: downloadsProject,
+      spec: 'simple_passing_spec.ts',
+      config: {
+        trashAssetsBeforeRuns: false,
+      },
+    })
+
+    const filePath = path.join(downloadsProject, 'cypress', 'downloads', 'records.csv')
+    const exists = await fs.pathExists(filePath)
```

Comment:
is there anywhere that the contents of the download are asserted on? maybe worth adding a test if not

Reply:
They're asserted on in the integration test itself.

Reply:
oh i see now https://github.com/cypress-io/cypress/blob/TR-602-clear-downloads/packages/server/test/support/fixtures/projects/downloads/cypress/integration/downloads_spec.ts